helm-charts/basehub/templates/dex/secret.yaml

```diff
@@ -0,0 +1,21 @@
+{{- if .Values.dex.enabled -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: dex
+type: Opaque
+stringData:
+  {{- $k8sState := lookup "v1" "Secret" .Release.Namespace "dex" | default (dict "data" (dict)) }}
+
+  {{- if hasKey $k8sState.data "oauth2Proxy.clientSecret" }}
+    oauth2Proxy.clientSecret: {{ index $k8sState.data "oauth2Proxy.clientSecret" | b64dec }}
+  {{- else }}
+    oauth2Proxy.clientSecret: {{ include "randHex" 64  }}
+  {{- end }}
+
+  {{- if hasKey $k8sState.data "oauth2Proxy.cookieSecret" }}
+    oauth2Proxy.cookieSecret: {{ index $k8sState.data "oauth2Proxy.cookieSecret" | b64dec }}
+  {{- else }}
+    oauth2Proxy.cookieSecret: {{ include "randHex" 16 }}
+  {{- end }}
```

Comment:
It may read clearer in this case, but it can also encourage this kind of practice to indent content relative to logic in helm templates.

+1 for consistent use of indentation practice, in this case it would mean to dedent line 11, 13, and 19 two spaces.

Reply:
Done